pkg/pillar/cmd/diag/diag.go

```diff
@@ -112,7 +112,11 @@ func Run() {
 	ctx.DevicePortConfigList = &types.DevicePortConfigList{}
 
 	// Make sure we have a GlobalConfig file with defaults
-	types.EnsureGCFile()
+	pubGlobalConfig, err := pubsub.PublishPersistent("", types.GlobalConfig{})
```

Comment:
Why is Diag publishing GlobalConfig?? Shouldn't Diag wait till GlobalConfig is published by zedagent?

Reply:
Belt and suspenders to make sure that we have a GlobalConfig file. We do this in the agents which run early, which includes nim, diag.

Note that this PR doesn't change that. Just restructures the code to make it easier for Avi.